x-pack/plugins/cases/server/services/attachments/index.ts

```diff
@@ -279,4 +285,105 @@ export class AttachmentService {
       throw error;
     }
   }
+
+  public async getCaseCommentStats({
+    unsecuredSavedObjectsClient,
+    caseIds,
+  }: {
+    unsecuredSavedObjectsClient: SavedObjectsClientContract;
+    caseIds: string[];
+  }): Promise<Map<string, CommentStats>> {
+    if (caseIds.length <= 0) {
+      return new Map();
+    }
+
+    interface AggsResult {
+      references: {
+        caseIds: {
+          buckets: Array<{
+            key: string;
+            doc_count: number;
+            reverse: {
+              comments: {
+                buckets: {
+                  alerts: {
+                    doc_count: number;
+                  };
+                  nonAlerts: {
+                    doc_count: number;
+                  };
+                };
+              };
+            };
+          }>;
+        };
+      };
+    }
+
+    const res = await unsecuredSavedObjectsClient.find<unknown, AggsResult>({
+      hasReference: caseIds.map((id) => ({ type: CASE_SAVED_OBJECT, id })),
+      hasReferenceOperator: 'OR',
+      type: CASE_COMMENT_SAVED_OBJECT,
+      perPage: 0,
+      aggs: AttachmentService.buildCommentStatsAggs(caseIds),
+    });
+
+    return (
+      res.aggregations?.references.caseIds.buckets.reduce((acc, idBucket) => {
+        acc.set(idBucket.key, {
+          nonAlerts: idBucket.reverse.comments.buckets.nonAlerts.doc_count,
+          alerts: idBucket.reverse.comments.buckets.alerts.doc_count,
+        });
+        return acc;
+      }, new Map<string, CommentStats>()) ?? new Map()
+    );
+  }
+
+  private static buildCommentStatsAggs(
+    ids: string[]
+  ): Record<string, estypes.AggregationsAggregationContainer> {
+    return {
+      references: {
+        nested: {
+          path: `${CASE_COMMENT_SAVED_OBJECT}.references`,
+        },
+        aggregations: {
+          caseIds: {
+            terms: {
+              field: `${CASE_COMMENT_SAVED_OBJECT}.references.id`,
+              // +1 to make sure we request more than the number of ids that were asked for
```

Comment:
Why that?

Reply:
Yeah I'll just change this back to the size of the `ids`.